flaky/_flaky_plugin.py

```diff
@@ -94,6 +104,32 @@ def _handle_test_error_or_failure(self, test, err):
             self._log_test_failure(name, err, message)
             return False
 
+    def _should_rerun_test(self, test, name, err):
+        """
+        Whether or not a test should be rerun.
+        This is a pass-through to the test's rerun filter.
+
+        :param test:
+            The test that has raised an error
+        :type test:
+            :class:`nose.case.Test` or :class:`Function`
+        :param name:
+            The test name
+        :type name:
+            `unicode`
+        :param err:
+            Information about the test failure (from sys.exc_info())
+        :type err:
+            `tuple` of `class`, :class:`Exception`, `traceback`
```

Comment:
These docstring types are getting tough to keep up with, I agree.

The problem is that Nose and Pytest each wrap a lot of things in their own classes, so many types have many possibilities, and the docstrings don't currently do a great job of making that clear.

This param info was copied from other code that use similar parameters.

I think a future PR is needed to clean this up across the codebase.